times/times.go

```diff
@@ -8,30 +8,19 @@ package times
 
 import (
 	"context"
-	"math"
 	"runtime"
 	"sort"
 	"sync/atomic"
 	"time"
 
-	log "github.com/sirupsen/logrus"
-	"golang.org/x/sys/unix"
-
 	"github.com/elastic/otel-profiling-agent/periodiccaller"
-	"github.com/elastic/otel-profiling-agent/util"
 )
 
 const (
 	// Number of timing samples to use when retrieving system boot time.
 	sampleSize = 5
-	// In the kernel, we retrieve timestamps from a monotonic clock
-	// (bpf_ktime_get_ns) that does not count system suspend time.
-	// In userspace, we try to detect system suspend events by diffing
-	// with values retrieved from a monotonic clock that does count
-	// the system suspend time.
-	// If the delta exceeds the following threshold (10s), we add it to the
-	// monotonic 'delta' that we keep track of.
-	monotonicThresholdNs = 10 * 1e9
+	// How often to attempt a sync with the realtime clock.
+	realtimeSyncInterval = 20 * time.Minute
```

Comment:
There are multiple options for realtime clock sync:

1. Sync once, on agent startup (calculate the delta once which remains fixed for the life of the agent process).
2. Sync periodically but do not allow time to drift backwards (only update delta if `newDelta - oldDelta > 0`)
3. Sync periodically (this PR).

Option 3 is more accurate in reflecting realtime clock changes but can result in discontinuous (non-monotonic) timestamps.

Reply:
Can you amend this to 2 or 3 minutes and make it configurable, following your suggestion? For me it's fine if we do the latter in a follow-up PR.